src/components/ServiceDetails/index.js

```diff
@@ -108,18 +110,36 @@ class ServiceDetails extends Component {
       createModelCalled: "edit",
       modelDetailsOnEdit: model,
     });
+    
   };
 
   onCancelEditModel = () => {
     this.setState({ activeTab: 0 })
   }
 
-  onUpdateModel = () => {
+  onUpdateModel = async(updateModelParams) => {
+    const { org_id, service_id } = this.props.service;
+      const sdk = await initSdk();
+      const serviceClient = new ServiceClient(sdk, org_id, service_id, sdk._mpeContract, {}, this.props.groupInfo);
+      const params = {
+        modelId: this.state.modelDetailsOnEdit.modelId,
+        address: this.props.wallet.address,
+        method: this.state.modelDetailsOnEdit.methodName,
+        name: this.state.modelDetailsOnEdit.serviceName,
+        modelName:updateModelParams.trainingModelName,
+        description:updateModelParams.trainingModelDescription,
+        publicAccess:updateModelParams.enableAccessModel,
+        addressList:updateModelParams.ethAddress,
+        status:this.state.modelDetailsOnEdit.status,
+        updatedDate:this.state.modelDetailsOnEdit.updatedDate,
+      };
+      const response=await serviceClient.updateModel(params);
+      console.log("=====updatingModel==", response);
     this.setState({ activeTab: 0 })
   }
 
   render() {
-    const { classes, service, pricing, loading, error, history, groupInfo, match, training, isLoggedIn } = this.props;
+    const { classes, service, pricing, loading, error, history, groupInfo, match, training, isLoggedIn,wallet} = this.props;
```

Comment:
Are you using wallet anywhere in this page ?

Reply:
Yes, for the address one